samples/contrib/kubeflow-e2e-mnist/kubeflow-e2e-mnist-kfpv2.ipynb

```diff
@@ -0,0 +1,777 @@
+    "            },\n",
+    "            \"spec\": {\n",
+    "                \"containers\": [\n",
+    "                    {\n",
+    "                        \"name\": \"tensorflow\",\n",
+    "                        \"image\": \"docker.io/liuhougangxa/tf-estimator-mnist\",\n",
+    "                        \"command\": [\n",
+    "                            \"sh\",\n",
+    "                            \"-c\"\n",
+    "                        ],\n",
+    "                        \"args\": [\n",
+    "                            \"python /opt/model.py --tf-export-dir=/mnt/export --tf-train-steps={} {}\".format(training_steps, best_hps)\n",
+    "                        ],\n",
+    "                        \"volumeMounts\": [\n",
+    "                            {\n",
+    "                                \"mountPath\": \"/mnt/export\",\n",
+    "                                \"name\": \"model-volume\"\n",
+    "                            }\n",
+    "                        ]\n",
+    "                    }\n",
+    "                ],\n",
+    "                \"volumes\": [\n",
+    "                    {\n",
+    "                        \"name\": \"model-volume\",\n",
+    "                        \"persistentVolumeClaim\": {\n",
+    "                            \"claimName\": model_volume_name\n",
+    "                        }\n",
+    "                    }\n",
+    "                ]\n",
+    "            }\n",
+    "        }\n",
+    "    }\n",
+    "\n",
+    "    tfjob_worker_spec = {\n",
+    "        \"replicas\": 1,\n",
+    "        \"restartPolicy\": \"OnFailure\",\n",
+    "        \"template\": {\n",
+    "            \"metadata\": {\n",
+    "                \"annotations\": {\n",
+    "                    \"sidecar.istio.io/inject\": \"false\"\n",
+    "                }\n",
+    "            },\n",
+    "            \"spec\": {\n",
+    "                \"containers\": [\n",
+    "                    {\n",
+    "                        \"name\": \"tensorflow\",\n",
+    "                        \"image\": \"docker.io/liuhougangxa/tf-estimator-mnist\",\n",
+    "                        \"command\": [\n",
+    "                            \"sh\",\n",
+    "                            \"-c\",\n",
+    "                        ],\n",
+    "                        \"args\": [\n",
+    "                          \"python /opt/model.py --tf-export-dir=/mnt/export --tf-train-steps={} {}\".format(training_steps, best_hps) \n",
+    "                        ],\n",
+    "                    }\n",
+    "                ],\n",
+    "            }\n",
+    "        }\n",
+    "    }\n",
+    "    \n",
+    "    # Convert chief spec to String type Output.\n",
+    "    import json\n",
+    "    chief_spec_str = json.dumps(tfjob_chief_spec)\n",
+    "    \n",
+    "    # Convert worker spec to String type Output.\n",
+    "    import json\n",
+    "    worker_spec_str = json.dumps(tfjob_worker_spec)\n",
```

Comment:
Do you think it makes sense to return this (and `tfjob_chief_spec`) as a dict, letting the lightweight component backend handle the serialization/deserialization?

Reply:
Thank you for the comment! I like the idea of using dict. However, These two values are needed in the container component below: `tfjob_launcher`. If I understand it correctly, I have to convert Dict to str for container component input. Correct?

Reply:
You can pass the dict directly to the container component. The conversion to a string happens implicitly when the commands/args are provided at container runtime.

As a minimal example, the following is a valid pipeline:

```python
from kfp import dsl
from typing import Dict


@dsl.component
def comp1() -> Dict[str, str]:
    return {'string': 'string'}


@dsl.container_component
def container(dictionary: Dict[str, str]):
    return dsl.ContainerSpec(
        image='alpine',
        command=['echo', dictionary],
    )


@dsl.pipeline
def my_pipeline():
    op1 = comp1()
    op2 = container(dictionary=op1.output)
```

Reply:
It is nice to know! Thank you Connor and I am able to make use of the said feature to convert Dict to str.